classes/Dispatcher.php

```diff
@@ -1184,4 +1184,35 @@ public static function getControllersInDirectory($dir)
 
         return $controllers;
     }
+
+    /**
+     * Get names of all available FO controllers.
+     *
+     * @var mixed
+     *
+     * @return array
+     */
+    public static function getControllersNames($dirs)
+    {
+        if (!is_array($dirs)) {
+            $dirs = [$dirs];
+        }
+
+        $controllers = Dispatcher::getControllers($dirs);
+
+        $controllersNames = [];
+        foreach ($controllers as $controller) {
+            $reflectionClass = new \ReflectionClass($controller);
```

Comment:
Reflection is slow but if `getControllersNames()` is called during hook setup (and not at runtime, when FO page with hooks on it is loaded) it's OK 👍

Reply:
https://ocramius.github.io/blog/accessing-private-php-class-members-without-reflection/

Reply:
@Matt75 In my opinion, the way described in the blog post you've shared sounds very difficult to read and so to maintain. Like @matks says, this function is executed only during hook setup, not on runtime. Do we sacrificing readability for a one time performance gain ?

Reply:
Nope, readability is more important ;)